htdocs/core/lib/security.lib.php

```diff
@@ -1052,6 +1052,9 @@ function checkUserAccessToObject($user, array $featuresarray, $object = 0, $tabl
 				if ($user->socid != $objectid) {
 					return false;
 				}
+			} elseif (isModEnabled('societe') && !($user->hasRight('societe', 'lire') && !$user->hasRight('societe', 'client', 'voir'))) {
```

Comment:
I think what you want to do is:
`} elseif (isModEnabled('societe') && !$user->hasRight('societe', 'lire') && !$user->hasRight('societe', 'client', 'voir')) {`
So: if thirdparty module is used, and user has no permisstion to read and no permission to read all (what you call "no permission of any kind", the permission check is false.
If you keep the parenthesis, the test become if has no permission to read OR can read all.
!( a and b)   is   (!a or !b)